### ISSUE TITLE
Disable adding a shortcut on the desktop

### ISSUE THREAD

**Issue by D0J0P** [microsoft/winget-cli#1545]
# Brief description of your issue
When upgrading packages using Winget, it adds a new shortcut to my desktop when I didn't have one already.

# Steps to reproduce
Open Cmd, type Winget Upgrade [-id of package or --all]
Watch new desktop shortcuts being created on your desktop.

# Expected behavior
That it wouldn't create a new desktop shortcut for programs I already have installed. 

# Actual behavior
Winget adds a new desktop shortcut after upgrading packages automatically.

# Environment
```none
winget --info
Windows Package Manager v1.1.12653
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19042.1237
Package: Microsoft.DesktopAppInstaller v1.16.12653.0
```

**Comment by Masamune3210** [comment]
Unfortunately unless a specific installer has a flag for not creating the shortcut there might not be much winget can do

**Comment by jedieaston** [comment]
What app are you experiencing this with @D0J0P? Maybe we can fix it.

**Comment by denelon** [comment]
@D0J0P I believe @Masamune3210 is correct. I'll keep this feature open to help with duplicate requests. We could also consider settings for this in the future, but we will likely still be depending on the installer.

**Comment by sredna** [comment]
> This is something that will certainly take time, and need to evolve.

It is a bit of a chicken/egg problem. I'd be happy to update the examples we (NSIS/nullsoft) ship with whatever is decided is the optimal solution but I also don't want to lead people down the wrong path prematurely.

In its simplest form it might just be `if (thisVer > existingVer) default_shortcuts_to_unchecked();`, this would at least handle the case where a user re-installs the same version to repair a broken install assuming creating shortcuts is defined as part of a repair operation.

End users also have the option of marking the .lnk files as hidden, this should work for most people unless you have the show hidden files option on.

**Comment by GlassedSilver** [comment]
Just check the items on the desktop for all users and the currently logged in user, diff to post-install and then revert changes.

Boom. Done. No developer sanity required.

**Comment by jedieaston** [comment]
I think the reason we can’t do that is because of the user saves a document to the desktop during a install we won’t know who did it necessarily. 

If we could figure out what was there pre and post install and then look for .lnk files that point to directories modified with the installer, then we’d be on to something. But I don’t know if the “track installed packages” feature is planning to be that deep.

**Comment by GlassedSilver** [comment]
Right, well... As you mentioned that'd be something that sounds plausible to do in addition.

I think `.lnk`s getting yote if they point to a dir that is home to applications in general could be a worthwhile intermediary option.

Additionally I'd propose to offer a standard for pre-defining certain common options for an unattended silent installer to follow.

I'm sure if winget defines default values for certain commonly expected user inputs the installer could then follow them or if the installer isn't ready to follow them yet alert the user and ask what to do. (proceed without args or cancel)

Because I'm not eager to try out random installers and going with the default setup only to figure out later on that the default setup includes today's equivalent to browser toolbars and Bonzi Buddy if you get my drift.

Cause the whole thing about shortcuts on the desktop is just the tip of the iceberg.

If winget would push for substantial change in how installers on Windows operate in this way in general this would be a MASSIVE gain for Windows in general.

**Comment by Jackaed** [comment]
Can we please push for this? This issue is the biggest annoyance I have with Winget at the moment.

**Comment by xzenor** [comment]
how about just giving the option at least for those package setups that do support it. afaik there's no way at all right now so even if there is a no-desktop-icon option in the setup, it can't be used. And just ignore it for the setups that don't support it.

**Comment by AndreasBrostrom** [comment]
I also want this feature.

I threw together two ugly tiny wrappers based on my winget/scoop/choco/wsl-dist mix system wide upgrade script.
What it does is that it scan the desktop first and after clears any new item created when done. (So don't save a document on your desktop while updateing/installing)
```pwsh
# PowerShell 7.3.7
function winget-install {
    $Desktops =    "$env:USERPROFILE\Desktop\$ShortcutName",
    "C:\Users\Default\Desktop\$ShortcutName",
    "C:\Users\Public\Desktop\$ShortcutName" 
    $preDesktop = @()
    foreach ($Desktop in $Desktops) {
        $items = Get-ChildItem -Path $Desktop -Name -Include "*.lnk"
        foreach ($item in $items) {
            $preDesktop += $item
        }
    }
    
    # Update WinGet
    winget install $args 
    
    # Cleaning up new unwhanted desktop icons
    Write-Host "Cleaning up WinGet created desktop icons...`n"
    $newDesktopLinks = @()
    foreach ($Desktop in $Desktops) {
        $items = Get-ChildItem -Path $Desktop -Name -Include "*.lnk"
        foreach ($item in $items) {
            if ($preDesktop -contains $item ) {
            } else {
                $newDesktopLinks += $item
            }
        }
    }
    foreach ($item in $newDesktopLinks) {
        Remove-Item -Path "$env:USERPROFILE\Desktop\$item" >$null 2>&1
        Remove-Item -Path "C:\Users\Default\Desktop\$item" >$null 2>&1
        Remove-Item -Path "C:\Users\Public\Desktop\$item" >$null 2>&1
        Write-Host "Cleaned up $item" -ForegroundColor DarkGray
    }
}

function winget-upgrade {
    $Desktops =    "$env:USERPROFILE\Desktop\$ShortcutName",
    "C:\Users\Default\Desktop\$ShortcutName",
    "C:\Users\Public\Desktop\$ShortcutName" 
    $preDesktop = @()
    foreach ($Desktop in $Desktops) {
        $items = Get-ChildItem -Path $Desktop -Name -Include "*.lnk"
        foreach ($item in $items) {
            $preDesktop += $item
        }
    }
    
    # Update WinGet
    winget upgrade $args
    
    # Cleaning up new unwhanted desktop icons
    Write-Host "Cleaning up WinGet created desktop icons...`n"
    $newDesktopLinks = @()
    foreach ($Desktop in $Desktops) {
        $items = Get-ChildItem -Path $Desktop -Name -Include "*.lnk"
        foreach ($item in $items) {
            if ($preDesktop -contains $item ) {
            } else {
                $newDesktopLinks += $item
            }
        }
    }
    foreach ($item in $newDesktopLinks) {
        Remove-Item -Path "$env:USERPROFILE\Desktop\$item" >$null 2>&1
        Remove-Item -Path "C:\Users\Default\Desktop\$item" >$null 2>&1
        Remove-Item -Path "C:\Users\Public\Desktop\$item" >$null 2>&1
        Write-Host "Cleaned up $item" -ForegroundColor DarkGray
    }
}
```
In action:
```pwsh
~ 
PS > pwsh -v
PowerShell 7.3.7
~ 
PS > winget-install Discord.Discord --force
Found Discord [Discord.Discord] Version 1.0.9018
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Starting package install...
Successfully installed
Cleaning up WinGet created desktop icons...

Cleaned up Discord.lnk
~ took 10s
PS >
````

**Comment by Pascal-KOTTE** [comment]
I join this request, it is just a surprise not able to have this kind of feature by design, else changing a registry somewhere....

Would be nice if able to add a global settings in the Winget settings file: So we can deactivate the default desktop shortcuts.

    "$schema": "https://aka.ms/winget-settings.schema.json",

    // For documentation on these settings, see: https://aka.ms/winget-settings
    // "source": {
    //    "autoUpdateIntervalInMinutes": 5
    // },
}

**Comment by jantari** [comment]
@AndreasBrostrom very nice, but it can be simplified a lot:

```powershell
# PowerShell 7.3.7
function winget-install {
    $preDesktop = [Environment]::GetFolderPath('Desktop'), [Environment]::GetFolderPath('CommonDesktop') |
        Get-ChildItem -Filter '*.lnk'
    
    # install WinGet
    winget install $args 

    $postDesktop = [Environment]::GetFolderPath('Desktop'), [Environment]::GetFolderPath('CommonDesktop') |
        Get-ChildItem -Filter '*.lnk'    

    # Cleaning up new unwhanted desktop icons
    Write-Host "Cleaning up WinGet created desktop icons..."
    $postDesktop | Where-Object FullName -notin $preDesktop.FullName | Foreach-Object {
        Remove-Item -LiteralPath $_.FullName
        Write-Host "Cleaned up $($_.Name)" -ForegroundColor DarkGray
    }
}

function winget-upgrade {
    $preDesktop = [Environment]::GetFolderPath('Desktop'), [Environment]::GetFolderPath('CommonDesktop') |
        Get-ChildItem -Filter '*.lnk'
    
    # Update WinGet
    winget upgrade $args

    $postDesktop = [Environment]::GetFolderPath('Desktop'), [Environment]::GetFolderPath('CommonDesktop') |
        Get-ChildItem -Filter '*.lnk'    

    # Cleaning up new unwhanted desktop icons
    Write-Host "Cleaning up WinGet created desktop icons..."
    $postDesktop | Where-Object FullName -notin $preDesktop.FullName | Foreach-Object {
        Remove-Item -LiteralPath $_.FullName
        Write-Host "Cleaned up $($_.Name)" -ForegroundColor DarkGray
    }
}
```

This will also work for users who have their desktop redirected to a non-standard path (e.g. because they're using OneDrive) because it isn't hardcoded to be at the _default_ location `$env:USERPROFILE\Desktop`.

**Comment by bitkris-dev** [comment]
Any update on this? I'm new on WingetUI but it's starting to bother me a lot and it's sad since it's an awesome piece of software.

**Comment by hrsa** [comment]
we desperately need a global confing to override this behaviour, like:
```
[shortcuts]
startmenu=0
desktop=0
explorer=0
quicklaunch=0
```

**Comment by Pascal-KOTTE** [comment]
> Any update on this? I'm new on WingetUI but it's starting to bother me a lot and it's sad since it's an awesome piece of software.
WingetUI ? from https://github.com/marticliment/wingetui ??
You are not on the good place !

**Comment by Dracrius** [comment]
Gosh darn it reading this thread breaks my heart how is this STILL an issue. Please just add a nuclear option to delete all shortcuts after updates or better yet just have it cache the list of shortcuts on the desktop before updates and then a minute after the last update finishes do a clean up! This really is a disgusting side effect for an updater as who in their right mind has an icon for literally everything on their desktop. I think more things I update add icons then things that don't and worst of all is I KNOW many of these apps don't auto add icons when using the in app update methods or installers which ever they offer.

**Comment by TK1987** [comment]
@jantari Why not using FileSystemWatchers instead? This would remove the desktop icons immediately after they have been created:
```
function winget-install {
  # Register events for removing desktop shortcuts
  foreach ($dir in "Desktop","CommonDesktop") {
    $Watcher = [System.IO.FileSystemWatcher]::new([Environment]::GetFolderPath($dir),"*.lnk")
    [void](Register-ObjectEvent -InputObject $Watcher -EventName "Created" -SourceIdentifier $dir -Action {Remove-Item -Force $EventArgs.FullPath})
  }

  # Install WinGet
  winget install $args

  # Unregister Events
  foreach ($dir in "Desktop","CommonDesktop") {Unregister-Event -SourceIdentifier $dir}
}
```

**Comment by MrAbidPatel** [comment]
I'd like this feature too. Firefox, Teamviewer and Ubisoft Connect are 3 culprits that I have installed that will always add a desktop shortcut on upgrade.

**Comment by popy2k14** [comment]
Same request here.
Such a feature would be nice to not clutter the desktop.

I think also a workaround (like the powershell script above) but implemented in winget would be nice.

thx

**Comment by whindsaks** [comment]
Actual guidance from the Windows team would be nice. Only Microsoft has the power to standardize ARP entries like NoDesktopShortcut etc.

**Comment by Masamune3210** [comment]
Microsoft has actually been quite hands-off when it comes to installers for the most part, historically. That's half the reason we are in this mess to begin with.

Also, adding such a entry to the ARP key wouldn't solve anything, it would need to be an argument passed into the installer, since WinGet handles upgrades by downloading and launching the installer from the product's website and letting the installer handle the upgrade with its own logic.

**Comment by whindsaks** [comment]
> Also, adding such a entry to the ARP key wouldn't solve anything

It would, it would encourage developers to add support for this feature just like people did for DisplayIcon and all the other standard entries that were invented for Windows 2000.

It would be even better if Microsoft could standardize the switches used to disable shortcut creation but it is a bit too late for that. The MSDN description for "NoDesktopShortcut" could at least note that installers are encouraged to support a command line switch.

If NoDesktopShortcut gained wide support, WinGet could even set it before starting the upgrade. That would even work for installers without a command line switch.

(Creating a desktop shortcut goes against the Windows Logo guidelines for the past 25 years but nobody really cares about that it seems)

**Comment by GlassedSilver** [comment]
> Microsoft has actually been quite hands-off when it comes to installers for the most part, historically. That's half the reason we are in this mess to begin with.
> 
> Also, adding such a entry to the ARP key wouldn't solve anything, it would need to be an argument passed into the installer, since WinGet handles upgrades by downloading and launching the installer from the product's website and letting the installer handle the upgrade with its own logic.

You basically just said because Microsoft cannot effectively force developers to use it they shouldn't bother.

I don't know about you, but a lot of good has been brought to this world without force and mere encouraged standards.

Right now this is all a mess and nobody bothers to clean it up, it's clear that just sticking to status quo is giving up before even trying.

**Comment by objecttothis** [comment]
This conversation is unproductive.  Since the OP created this feature request the conversation has gone round and round between "please add this" to "can't, it's not designed well" to "you could do it like this" to "that wouldn't work" and then repeat.  The horse has been thoroughly beaten. Can we just leave it as a feature request and let it be implemented or not?

**Comment by Azertixe** [comment]
> What app are you experiencing this with @D0J0P? Maybe we can fix it.
VLC, LibreOffice, Audacity...

**Comment by totkeks** [comment]
> Unfortunately unless a specific installer has a flag for not creating the shortcut there might not be much winget can do

Which begs the question for me, why is there no standardized installer for Windows in all these years? MSI(X) is just a package format, right? It doesn't define the installer parameters? For this request to work, the installer would need to have a standardized way to enable/disable Desktop icons if applicable, right?

**Comment by excited-bore** [comment]
> > What app are you experiencing this with [@D0J0P](https://github.com/D0J0P)? Maybe we can fix it.
> 
> VLC, LibreOffice, Audacity...

I would like to add that Mozilla Thunderbird (and most likely also Firefox, but I haven't double checked as I there was always a firefox icon on my desktop while upgrading) and Albion Online also suffer from this.

**Comment by masterflitzer** [comment]
although i'd love a policy regarding this issue, winget is already struggling to establish itself, there's no way this would work, most app maintainers don't even care much about winget so they won't just respect this policy

**Comment by SirLouen** [comment]
> although i'd love a policy regarding this issue, winget is already struggling to establish itself, there's no way this would work, most app maintainers don't even care much about winget so they won't just respect this policy

Practically 99% of my apps are in Winget. I don't think there is an adoption issue here. All devs I've contacted in the past to add winget to their products, were willing to be compliant. As soon as they get the compliance warning (deprecation warning or whatever), they will update accordingly. It is not that hard.

**Comment by G0rocks** [comment]
Just to throw my opinion in here.

The methods suggested that would at least partly solve this problem should be implemented.

- There have already been posted methods that would work properly for most users for most cases (going through the shortcuts on the desktop before and after upgrades for example). Meaning there are solutions that would work (even though not perfectly).
- In my experience it's better to have something that mostly works than nothing that also doesn't work.
- This issue was posted 2021-10-03 and today is the 2025-05-04 (happy star wars day btw), the first suggested solution (that again would do something rather than nothing) was posted 2021-10-22 only 19 days after the original post and here we are 3 years and 7 months later and nothing has happened.

With a "constant development" mindset, a lot of progress could have been made over the last 3 years to have this feature working properly. Only if it was implemented to begin with and then for the "kinks" some other issues would have been posted in the last 3 years and 7 months and it would have been possible to work them out so with a "constant development" mindset, this issue would probably have already been implemented really well today.

In my opinion, let's implement the shortcuts created on desktop scraper that removes them if they weren't there before and let's implement the policy for developers to add a flag about these things and on the 4th of december 2028 (3 years and 7 months from now) we'll definitely have something that works well. Unless you'd rather manually delete all automatically created desktop shortcuts for the next 3 years and 7 months, it's up to you in the end.

**Comment by yllekz** [comment]
So many apps incorrectly place a desktop shortcut during silent install. Please add code to just remove an .lnk file created by the run. It shouldn't be difficult to do at all, and can just be an additional switch option.

**Comment by denelon** [comment]
WinGet doesn't have a way to know what other things are happening at the same time. Just naively removing .lnk files would likely end up with some frustrated users who don't understand what's happening.

Fortunately, this issue has been working its way up the ranks to the top of the "thumbs up" list. I'm expecting we'd want an argument and settings for the behavior. WinGet would most likely default to "no new icons" on the desktop (start menu, taskbar, etc.) during the upgrade flow, and for those packages with support, we'd be able to add specific flags, but that comes at the cost of a more complex manifest.

**Comment by Frulfump** [comment]
> WinGet doesn't have a way to know what other things are happening at the same time. Just naively removing .lnk files would likely end up with some frustrated users who don't understand what's happening.
> 
> Fortunately, this issue has been working its way up the ranks to the top of the "thumbs up" list. I'm expecting we'd want an argument and settings for the behavior. WinGet would most likely default to "no new icons" on the desktop (start menu, taskbar, etc.) during the upgrade flow, and for those packages with support, we'd be able to add specific flags, but that comes at the cost of a more complex manifest.

An idea would be if they had the link before the upgrade starts it stays and if they don't it gets removed. So would need a check if the lnk already exists?

But I think you should help manifest creators since many don't know that the installer that's used some times supports not adding the shortcut in the first place. At least that would tie us over until this feature is rolled out. Example from PR I created https://github.com/microsoft/winget-pkgs/pull/243547


Sounds like a good default, and I expect the added complexity being added would be minor (but if it isn't it's still worth it)

I also don't want desktop shortcuts or start menu entries, or taskbar modification on install either.

**Comment by whindsaks** [comment]
Please standardize a registry ARP value for ISVs to use in addition to guidelines for command line arguments. "NoDesktopLink", DesktopLnk=0, "NoDesktopShortcutOnUpgrade" or whatever the OS team wants, just give us a name vendors can use.

**Comment by nickhomelab** [comment]
For those who don't mind using a GUI for their Winget operations, UniGetUI added an "Automatic desktop shortcut remover" feature earlier this year.

https://github.com/marticliment/UniGetUI/issues/3282

**Comment by G0rocks** [comment]
Another option is an idea I had just now which is for people (like me) who don't want any shortcuts on the desktop is to add a setting which is to delete all shortcuts from desktop after updates/installs without any checks for what they are for. That would be good enough for me at least and sounds simple to implement. Specifically because it's a setting that explicitly states it will delete all shortcuts, not only installed packages or updated ones.
I still think the idea in my previous comment is better but I just had this idea so I'm posting it also.

**Comment by whindsaks** [comment]
> delete all shortcuts from desktop after updates/installs

This would be covered by the "delete new shortcuts added during winget child process execution" option.

**Comment by G0rocks** [comment]
Just to be clear because I feel like we're misunderstanding each other perhaps.
That option sounds like a more sophisticated solution than what my idea was.
My idea is literally to remove 100% of all shortcuts on desktop regardless of why or when they showed up to desktop.
Example:
1. I have a shortcut to internet explorer on desktop.
2. I update winget using winget.
3. Shortcut to internet explorer on desktop is now gone because it was deleted because winget updated some package

It's very crude (MVP) but it would be a good enough solution for me :)

**Comment by SirLouen** [comment]
> Just to be clear because I feel like we're misunderstanding each other perhaps. That option sounds like a more sophisticated solution than what my idea was. My idea is literally to remove 100% of all shortcuts on desktop regardless of why or when they showed up to desktop. Example:
> 
> 1. I have a shortcut to internet explorer on desktop.
> 2. I update winget using winget.
> 3. Shortcut to internet explorer on desktop is now gone because it was deleted because winget updated some package
> 
> It's very crude (MVP) but it would be a good enough solution for me :)

`winget` doesn't interact with the desktop icons. Are the installers who actually do so. This is why, we are asking to force a policy for the correct management of Desktop Icons by the installers.

**Comment by G0rocks** [comment]
Yeah I know winget does not interact with desktop icons currently.
The idea I had this morning is to add a setting that deletes all shortcuts from desktop after every update or install from winget.
If that setting was implemented and winget would interact with desktop icons then that would solve this issue for me completely because I like my desktop to have zero shortcuts for anything. I usually assume also that if something applies to me it probably applies to somebody else so potentially this MVP would be good enough for a lot of people.

And as per my previous comment, getting something done is better than getting nothing done and then you can always improve. That includes the policies and such.

Though also, I'm just realising, having a package manager means that developers of programs don't need to program in an updater since the package manager can keep it up to date and that means that programs take less space on the hard drive and less work is spent in the world to make programs. The same applies to shortcuts on the desktop. Meaning that if winget would handle the desktop icons then the software for desktop icons would be handled in the package manager and no other program would technically need to make it or handle it. Which would also be cool in my opinion. But this is out of scope of this issue.

**Comment by protected22** [comment]
Yep I have the same problem. When I run _"winget upgrade --all". It also opens all installation-popups and for 90% of the programs also adds a new shortcut icon on my desktop.

**Comment by G0rocks** [comment]
Ok, I ended up just making a thing to deal with this.
I made a program:
https://github.com/G0rocks/desktop_shortcut_remover/

It checks the desktop of the current user and the public desktop folder for shortcuts every day at 10:00 (10 in the morning for AM/PM peeps) and deletes all shortcuts.

Just finished submitting it to the winget-pkgs repo so it should be installable with winget at some point.
https://github.com/microsoft/winget-pkgs/pull/320398

Feel free to add issues and stuff to the repository for changes you would like to see though know I am a busy person and might not implement things or if I will I might do it very slowly.

Note also this is a very minimum working program. After installing/downloading you have to run it once (or add it to your startup programs so it's always there) and then to shut down the program you must open the task manager and end the task. Super not user friendly but hey it works :)

**Comment by yllekz** [comment]
> > Just to be clear because I feel like we're misunderstanding each other perhaps. That option sounds like a more sophisticated solution than what my idea was. My idea is literally to remove 100% of all shortcuts on desktop regardless of why or when they showed up to desktop. Example:
> > 
> > 1. I have a shortcut to internet explorer on desktop.
> > 2. I update winget using winget.
> > 3. Shortcut to internet explorer on desktop is now gone because it was deleted because winget updated some package
> > 
> > It's very crude (MVP) but it would be a good enough solution for me :)
> 
> `winget` doesn't interact with the desktop icons. Are the installers who actually do so. This is why, we are asking to force a policy for the correct management of Desktop Icons by the installers.

I'd argue the opposite. Winget directly results in desktop icon clutter being added. I don't see any other explanation. There should be an option to skip or cleanup newly created desktop icons. No other package manager has this issue, to my knowledge.

**Comment by dathbe** [comment]
> I'd argue the opposite. Winget directly results in desktop icon clutter being added. I don't see any other explanation. There should be an option to skip or cleanup newly created desktop icons. No other package manager has this issue, to my knowledge.

To be fair, Ninite also places the icons on the desktop.
On the other hand, you're right though that this is probably not a problem that can just be punted to the individual developers saying that they choose to install the shortcut.  Plenty of these apps will not place a desktop icon if you install them directly, so it does seem possible to set up winget to not place the icon either--at least for those.

**Comment by Jenso-DE** [comment]
With the paid version of Ninite there is an option to "disable" the shortcuts.

**Comment by G0rocks** [comment]
The desktopshortcutremover was added to the packages repo, install with
`winget install G0rocks.DesktopShortcutRemover`

**Comment by Azertixe** [comment]
> G0rocks.DesktopShortcutRemover

Hi I wanna test it. How do it works ?

**Comment by G0rocks** [comment]
> > G0rocks.DesktopShortcutRemover
> 
> Hi I wanna test it. How do it works ?

Just open the command line, run `winget install G0rocks.DesktopShortcutRemover` and then run it and add it to your startup apps. That's it.
After that, every day at 10:00 (your computer time) the program will delete all shortcuts on your desktop. Feel free to add issues for bugs or feature requests on the project github if it's not working for you or if you need it to behave differently.
https://github.com/G0rocks/desktop_shortcut_remover/issues

Edit:
It has been working great for me, I've not really noticed that there are any shortcuts and the two times that they have appeared, the day after they were gone :)
Let me know how it goes!

**Comment by G0rocks** [comment]
There is a new version of the program coming out when [this PR](https://github.com/microsoft/winget-pkgs/pull/334853) goes through which has the new feature of attempting to delete shortcuts automatically only seconds after the shortcut has been created.

**Comment by marcoruval** [comment]
I know there are some people that like having some shortcuts but not all so, what about adding a flag to winget upgrade command to "remember" current shortcuts on desktop at command invocation, then after each correct package upgrade it checks again for any newly created shortcut and delete it.

**Comment by denelon** [comment]
Duplicate of #1045

**Comment by beedell-roke** [comment]
Then, @denelon, shouldn't this be closed *as a duplicate*, rather than as not planned?
